firestore/swift/firestore-smoketest/ViewController.swift

```diff
@@ -755,19 +755,39 @@ class ViewController: UIViewController {
         let docRef = db.collection("cities").document("BJ")
 
         docRef.getDocument { (document, error) in
+            if let error = error {
+                print("Error retrieving document: \(error)")
+                return
+            }
+
+            // Construct a Result type to encapsulate deserialization errors or
+            // successful deserialization. Note that if there is no error thrown
+            // the value may still be `nil`, indicating a successful deserialization
+            // of a value that does not exist.
+            //
+            // There are thus three cases to handle, which Swift lets us describe
+            // nicely with built-in sum types:
+            //
+            //      Result
+            //        /\
+            //   Error  Optional<City>
+            //               /\
+            //            Nil  City
             let result = Result {
-                try document.flatMap {
-                    try $0.data(as: City.self)
-                }
+              try document?.data(as: City.self)
             }
             switch result {
             case .success(let city):
                 if let city = city {
```

Comment:
Redundant?

Reply:
This would result in a compiler error for a non-optional `city`.